### PR TITLE
[WIP] vim-patch:8.0.0{428,429,430,436,488,585}

### PR DESCRIPTION
**Pull request by janlazo**
**vim-patch:8.0.0428: git and hg see new files after running tests**

Problem:    Git and hg see new files after running tests. (Manuel Ortega)
Solution:   Add the generated file to .hgignore (or .gitignore). Delete the
resulting verbose file. (Christian Brabandt)  Improve dependency
on opt_test.vim.  Reset the 'more' option.
https://github.com/vim/vim/commit/65408f7dfbd4bb50d740a8b8f2171080571223f2

**vim-patch:8.0.0429: options test does not always test everything**

Problem:    Options test does not always test everything.
Solution:   Fix dependency for opt_test.vim.  Give a message when opt_test.vim
was not found.
https://github.com/vim/vim/commit/e8512d7fedbddd8d464b2c589e6ba5ab9a41bc78

**vim-patch:8.0.0430: options test fails or hangs on MS-Windows**

Problem:    Options test fails or hangs on MS-Windows.
Solution:   Run it separately instead of part of test_alot.  Use "-S" instead
of "-u" to run the script.  Fix failures.
https://github.com/vim/vim/commit/bb962262eb6b18d93161d7506317d215ae80bfe5

**vim-patch:8.0.0436: running the options test sometimes resizes the terminal**

Problem:    Running the options test sometimes resizes the terminal.
Solution:   Clear out t_WS.
https://github.com/vim/vim/commit/ee1dd1cdc1f9a07e53b792e8762f515213b10c8a

**vim-patch:8.0.0488: running tests leaves an "xxx" file behind**

Problem:    Running tests leaves an "xxx" file behind.
Solution:   Delete the 'verbosefile' after resetting the option.
https://github.com/vim/vim/commit/b4c5572e749063daeef30cfc2932bea1fd468165

**vim-patch:8.0.0585: test_options fails when run in the GUI**

Problem:    Test_options fails when run in the GUI.
Solution:   Also check the 'imactivatekey' value when the GUI is not running.
            Specify test values that work and that fail.
vim/vim@86e5792

I doubt this works or if most of the changes are required since nvim has options.lua and `test/unit/*.lua` but the code generator is in Vimscript so it's worth a try.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -35,6 +35,7 @@ tags
 /src/nvim/testdir/test.log
 /src/nvim/testdir/messages
 /src/nvim/testdir/viminfo
+/src/testdir/opt_test.vim
```
Comment (justinmk) [inline .gitignore]: it would be `/src/nvim/testdir/...` in our case.

File: src/nvim/testdir/Makefile [hunk 2]
```diff
@@ -22,6 +22,7 @@ SCRIPTS_DEFAULT = \
            test48.out             \
            test52.out             \
            test64.out             \
+           test60.out             \
```
Comment (justinmk) [inline src/nvim/testdir/Makefile]: This should not be needed.

File: src/nvim/testdir/Makefile [hunk 3]
```diff
@@ -183,6 +184,7 @@ clean:
 	        *.res          \
 	        *.rej          \
 	        *.orig         \
+	        opt_test.vim   \
```
Comment (justinmk) [inline src/nvim/testdir/Makefile]: don't need this

**Comment by janlazo** [comment]
I will leave this as is until these patches are required by other patches. Feel free to take it over in the meantime.

**Comment by janlazo** [comment]
Moving some patches here to https://github.com/neovim/neovim/pull/8605

**Comment by janlazo** [comment]
```vim
set cpo=&vim
```
Why does this work in Vim? `echo &cpo` outputs `&vim`. Looks wrong to me so I'll remove the `=`.

File: src/nvim/testdir/gen_opt_test.vim [hunk 4]
```diff
@@ -0,0 +1,211 @@
+" Script to generate testdir/opt_test.vim from option.c
+
+set cpo=&vim
```
Comment (janlazo) [inline src/nvim/testdir/gen_opt_test.vim]: `&` POSIX flag. Intentional?

File: src/nvim/testdir/gen_opt_test.vim [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+" The terminal size is restored at the end.
+" Clear out t_WS, we don't want to resize the actual terminal.
+let script = [
+      \ 'let save_columns = &columns',
+      \ 'let save_lines = &lines',
+      \ 'let save_term = &term',
+      \ 'set t_WS=',
+      \ ]
+
+/#define p_term
```
Comment (janlazo) [inline src/nvim/testdir/gen_opt_test.vim]: N/A for nvim

**Comment by janlazo** [comment]
Having seconds thoughs on this PR after running `gen_opt_test.vim` on Vim.
I posted `opt_test.vim` in https://gist.github.com/janlazo/90fc0df231afdf3edba9ab793c17416b.
I rather reuse `src/nvim/options.lua` on a unit test if there's no unit test for all nvim options.

**Comment by justinmk** [comment]
> I rather reuse src/nvim/options.lua on a unit test if there's no unit test for all nvim options

Sounds reasonable, it's also fine to just mark these patches as NA, we may already have pretty good coverage with the existing Lua tests.

**Pull request closed without merging**